Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/EditProjectUrlController.java

```diff
@@ -0,0 +1,91 @@
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+import edu.asu.spring.quadriga.validator.ProjectURLValidator;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+@Controller
+public class EditProjectUrlController {
+
+    @Autowired
+    private IModifyProjectManager projectManager;
+
+    @Autowired
+    private IRetrieveProjectManager retrieveProjectManager;
+
+    @Autowired
+    private ProjectURLValidator validator;
+
+    @Resource(name = "projectconstants")
+    private Properties messages;
+
+    private static final Logger logger = LoggerFactory.getLogger(ModifyProjectController.class);
+
+    /**
+     * Attach the custom validator to the Spring context
+     */
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+        binder.setValidator(validator);
+    }
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 1, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/editProjectPageURL/{projectid}", method = RequestMethod.GET)
+    public String editProjectPageURL(@PathVariable("projectid") String projectid, Principal principal, ModelMap model)
+            throws QuadrigaStorageException, QuadrigaAccessException {
+        IProject project = retrieveProjectManager.getProjectDetails(projectid);
+        model.addAttribute("project", project);
+        model.addAttribute("unixnameurl", messages.getProperty("project_unix_name.url"));
+        return "auth/workbench/editProjectPageURL";
+    }
+
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 3, userRole = {
+            RoleNames.ROLE_COLLABORATOR_ADMIN, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/editProjectPageURL/{projectid}", method = RequestMethod.POST)
+    public String editProjectPageURL(@Validated @ModelAttribute("project") Project project, BindingResult result,
+            @PathVariable("projectid") String projectid, Principal principal, ModelMap model,
+            RedirectAttributes redirectAttributes) throws QuadrigaStorageException, QuadrigaAccessException {
+        if (result.hasErrors()) {
+            logger.error("Update project details error:", result);
+            model.addAttribute("project", project);
+            model.addAttribute("unixnameurl", messages.getProperty("project_unix_name.url"));
+            return "auth/workbench/editProjectPageURL";
+        }
+        String userName = principal.getName();
+        projectManager.updateProjectURL(project.getProjectId(), project.getUnixName(), userName);
+        redirectAttributes.addFlashAttribute("show_success_alert", true);
+        redirectAttributes.addFlashAttribute("success_alert_msg", "Project URL has been updated successfully");
+        return "redirect:/auth/workbench?projectid=" + projectid;
```

Comment:
What does this redirect to? Shouldn't that redirect to the project page, which would be 
auth/workbench/projectid?